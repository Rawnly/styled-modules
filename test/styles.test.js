import _transform from './_transform';
import plugin from '../src/babel-external';

const transform = (file, opts) => _transform(file, {
  plugins: [plugin],
  ...opts,
});

it('works', async () => {
  const { code } = await transform('./fixtures/styles.fixture.js');
  expect(code).toMatchSnapshot();
});

it('generates source maps', async () => {
  const { code } = await transform('./fixtures/styles.fixture.js', {
    sourceMaps: true,
  });
  expect(code).toMatchSnapshot();
});
