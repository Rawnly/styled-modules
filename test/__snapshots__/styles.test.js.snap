// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates source maps 1`] = `
"module.exports = {
  helloWorld: 'helloWorld'
};

module.exports.__hash = '1910013378';
module.exports.__source = '\\\\n  .helloWorld {\\\\n    color: red;\\\\n  }\\\\n\\\\n/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlcy5maXh0dXJlLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUlBIiwiZmlsZSI6InN0eWxlcy5maXh0dXJlLmpzIiwic291cmNlc0NvbnRlbnQiOlsibW9kdWxlLmV4cG9ydHMgPSB7XG4gIGhlbGxvV29ybGQ6ICdoZWxsb1dvcmxkJyxcbn07XG5cbm1vZHVsZS5leHBvcnRzLl9fc291cmNlID0gYFxuICAuaGVsbG9Xb3JsZCB7XG4gICAgY29sb3I6IHJlZDtcbiAgfVxuYDtcbiJdfQ== */\\\\n/*@ sourceURL=styles.fixture.js */';"
`;

exports[`works 1`] = `
"module.exports = {
  helloWorld: 'helloWorld'
};

module.exports.__hash = '1910013378';
module.exports.__source = '\\\\n  .helloWorld {\\\\n    color: red;\\\\n  }\\\\n';"
`;
